rules/klayout/drc/testing/run_regression.py

```diff
@@ -193,26 +204,52 @@ def call_regression(rule_deck_path, path):
         writer = csv.writer(f, delimiter=',')
         writer.writerows(conc)
 
-    print(f"\n Total rules in {name_ext} for {path}: {len(conc)} \n")
-    print(f"{passed} passed rules ")
-    print(f"{failed} failed rules ")
-    print(f"{not_tested_counter} Not tested rules \n")
+    logging.info(f"========= Summary Report in {name_ext} for {path} =========")
+    logging.info(f"Total rules: {len(conc)}")
+    logging.info(f"{not_tested_c} not tested rules")    
+    logging.info(f"{passed} passed rules ")
+    logging.info(f"{known_issues} known_issues rules ")
+    logging.info(f"{failed} failed rules ")
+    
     t1 = time.time()
-
-    print(f'Execution time {t1 - t0} s')
+    logging.info(f'Execution time {t1 - t0} s')
+    logging.info(f"===============================================================")
+
+    if failed > 0:
+        logging.info("Some unit tests has failed. Failing regression:")
+        df = pd.read_csv(f'run_{x}_{name_ext}/conclusion.csv')
+        pd.set_option('display.max_columns', None)
+        pd.set_option('display.max_rows', None)
+        pd.set_option("max_colwidth", None)
+        pd.set_option('display.width', 1000)
+        logging.info("## Full report:")
+        print(df)
+
+        print("\n")
+        logging.info("## Only failed")
+        print(df[df["Status"] == "Fail"])
+
+        exit(1)
     return report
 
 if __name__ == "__main__":
 
-    sub_report = []
-    full_report = []
+    # logs format
+    logging.basicConfig(level=logging.DEBUG, format=f"%(asctime)s | %(levelname)-7s | %(message)s", datefmt='%d-%b-%Y %H:%M:%S')
+
+    # Initial values for DRC report
+    sub_report   = []
+    full_report  = []
     final_report = [["Rule_Name", "Status"]]
-    final_detailed_report = [["Rule_Name", "False_Postive", "False_Negative", "Total_Violations", "Not_Tested"]]
+    final_detailed_report = [["Rule_Name", "False_Postive", "False_Negative", "Total_Violations", "Not_Tested" , "Known_issues"]]
 
+    # Start of execution time 
     t0 = time.time()
 
+    # Reading docopt arguments
     args = docopt(__doc__)
 
+    # DRC switches definitions 
     switches = ''
```

Comment:
maybe maintain the switches as a list and join them before calling the command?

Reply:
@proppy we can do that but string concatenation is working as well. I would consider that minor enhancement.